ethereum/rpc/backend/backend.go

```diff
@@ -69,15 +73,26 @@ func NewEVMBackend(logger log.Logger, clientCtx client.Context) *EVMBackend {
 	}
 }
 
-// BlockNumber returns the current block number.
+// BlockNumber returns the current block number in chain state.
```

Comment:
I'd add a note to outline the problem of using the height from blockchain info

Reply:
done